repl/src/main/scala/org/apache/spark/repl/SparkILoop.scala

```diff
@@ -124,6 +141,26 @@ class SparkILoop(in0: Option[BufferedReader], out: JPrintWriter)
     super.replay()
   }
 
+  /**
+   * TODO: Remove `runClosure` when the support of Scala 2.11 is ended
+   * In Scala 2.12, we don't need to use `savingContextLoader` to execute the `body`.
+   * See `SI-8521 No blind save of context class loader` for detail.
+   */
+  private def runClosure(body: () => Boolean): Boolean = {
+    if (isScala2_11) {
+      val loader = Utils.classForName("scala.reflect.internal.util.ScalaClassLoader$")
+        .getDeclaredField("MODULE$")
+        .get(null)
```

Comment:
@viirya Out of my curiosity, since it's a static method, in theory, in the `invoke`, the first arg should be `null`. Why do you put the loader in? Also, this doesn't work in tests, and we get the method not found exception.

Reply:
Although it is a static method in Scala, it will be compiled to a non-static method in Java class. This class has a static member of the same type. The access of all static methods are through the static member.